add login check decorator

wenjzhu opened the pull request:
## Description
_Please provide a summary of the pull request and the issue it fixes. Please add necessary details, context, dependencies, explanation of when review is needed (see next section), etc._

how to test:
- after seeing the homepage without logging in, try to access the .../contribute/create url
- the expected behavior is to take back to login page, instead of showing the submit page 
- once we login in, then we should be able to see the submit page well

**Fixes #(add issue number here and remove parentheses)**

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Immediately
- [ ] Within a week
- [x] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [ ] I have updated the [CHANGELOG](../CHANGELOG.md).
- [ ] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

ywkim312 commented on contributions/catalog/controllers/auth.py:
```diff
@@ -0,0 +1,26 @@
+    Blueprint, flash, g, redirect, render_template, request, session, url_for, current_app
+)
+
+def login_required(view):
```
I am not really sure and I am not expert of it but is it better to make this as class? Or is it better to have def inside the def? I am just asking since I don't know too much about this.

wenjzhu replied:
I am not sure about this - let us test this bit more and I can make the change if we feel needed.

ywkim312 reviewed (commented):
How do I check the change or your code in the browser? Could you provide brief explanation of what to see outside the code, such as browser or something else?

wenjzhu commented:
> How do I check the change or your code in the browser? Could you provide brief explanation of what to see outside the code, such as browser or something else?

my bad i forgot to provide some instruction about this. 

1. after seeing the homepage without logging in, try to access the submit page url (.../contribute/create)
2. the expected behavior is to take back to login page, instead of showing the submit page
3. once we login in, then we should be able to see the submit page well.

ywkim312 commented:
This is what happened when I accessed localhost:5050/contribute/create without logging in 
![image](https://user-images.githubusercontent.com/12547573/114897376-c9416380-9dd6-11eb-810b-230aec626958.png)

ywkim312 commented:
When I went to the same page with logged in, it redirected to http://localhost:5050/profile page

sandeep-ps commented:
> When I went to the same page with logged in, it redirected to http://localhost:5050/profile page

@wenjzhu, please create a follow-up issue to fix this. /profile page URL does not seem to have the /contribute URL prefix. Thanks.

sandeep-ps commented:
@wenjzhu, are there other pages to which we can add the `@login_required` decorator? I thought these need to be added to have already logged in:

/logout
/profile

I wasn't sure about the ones below. What do you think?

/results
/submitted

wenjzhu commented:
> This is what happened when I accessed localhost:5050/contribute/create without logging in
> ![image](https://user-images.githubusercontent.com/12547573/114897376-c9416380-9dd6-11eb-810b-230aec626958.png)
> 
thanks, i think this is due to my typo in checking condition...and just fixed this

ywkim312 commented:
> This is what happened when I accessed localhost:5050/contribute/create without logging in
> ![image](https://user-images.githubusercontent.com/12547573/114897376-c9416380-9dd6-11eb-810b-230aec626958.png)
> 
> thanks, i think this is due to my typo in checking condition...and just fixed this

I still see the same error. However, If I go to localhost:5050 then it redirects to github login

wenjzhu commented:
> > This is what happened when I accessed localhost:5050/contribute/create without logging in
> > ![image](https://user-images.githubusercontent.com/12547573/114897376-c9416380-9dd6-11eb-810b-230aec626958.png)
> > thanks, i think this is due to my typo in checking condition...and just fixed this
> 
> I still see the same error. However, If I go to localhost:5050 then it redirects to github login

Could you pull again? I tested on my local and it has no key error after I changed the auth validation...

sandeep-ps commented:
@wenjzhu, it looks like redirect after login is not working correctly. Please look into this. Thanks.

wenjzhu commented:
I removed the login decorator for the profile page which seems ok to me. 
But still not sure the cause of the error Yongwook mentioned above since it seems not to appear in my local...maybe it is related to cookie/session? let me know if you guys see the same error again.

sandeep-ps commented:
> > > This is what happened when I accessed localhost:5050/contribute/create without logging in
> > > ![image](https://user-images.githubusercontent.com/12547573/114897376-c9416380-9dd6-11eb-810b-230aec626958.png)
> > > thanks, i think this is due to my typo in checking condition...and just fixed this
> > 
> > 
> > I still see the same error. However, If I go to localhost:5050 then it redirects to github login
> 
> Could you pull again? I tested on my local and it has no key error after I changed the auth validation...

@ywkim312, when going to http://localhost:5050/contribute/create from an incognito window, it's redirecting me to the GitHub login page. As suggested by @wenjzhu, could you please pull the latest and try again? Thanks.

ywkim312 reviewed (approved):
It worked as it should be

sandeep-ps commented:
@wenjzhu, please resolve the merge conflict. Thanks.
